avocado/core/test.py

```diff
@@ -133,7 +133,136 @@ def str_filesystem(self):
                                  % (self.str_uid, str(self)))
 
 
-class Test(unittest.TestCase):
+class BaseTestData(object):
```

Comment:
Well I don't see a reason for a `BaseTestData` class. It is not public and all tests inherit the `TestData` via `avocado.Test` class. So I agree the mix-in `TestData` makes sense, the `BaseTestData` base-class is IMO unnecessary and makes things more complicated.

Anyway I'll proceed as this can be tweaked later (if necessary)

Reply:
I agree there's clearly no functional need.  I see it as "code based documentation": if I was reading this code, I think I'd easily understand what's the point behind it.

Reply:
OK, yesterday I was probably a bit picky. I looked at it and saw the very same structure only now with lambdas, which made me a bit angry.

For documentation purposes the documentation is the best place. I don't see much point in describing unavailable interface. It's more important to focus on the actual available interface, which is also described later in `TestData` class. Anyway this is definitely not a deal-breaker...

Reply:
Dropping the use of BaseTestData.